Fix trailing ; in notmuch config breaking SentMailsFilter

TimQuelch opened the pull request:
The trailing semicolon resulted in the string split containing an empty string. When the notmuch
query is built later on this results in no messages being matched by the filter.

e.g.  in notmuch-config
```
primary_email=primary@domain.com
other_email=secondary@domain.com;tertiary@domain.com;
```

This fix ensures that the empty string is not added to the `my_addresses` set.

A potentially better fix would be to ensure that `notmuch_settings.get()` removes trailing ;

This fixes #265

GuillaumeSeren commented:
So, I think this patch should take care of all case like that (such in MeFilter).

Ideally the problem should vanish if the list in the notmuch-config did not contain this extra semicolon, but being sure that we don't generate blank item in list can not be wrong.

TimQuelch commented:
I've changed it to use the existing get_list function. This already filters out empty strings by default.

I also applied the fix to MeFilter, which was the only place that I could find that used the `.get(.*)\.split` pattern.

GuillaumeSeren commented:
Hey @TimQuelch ,
yes I aggree on the usage of the get_list function as it done on a similar case here https://github.com/afewmail/afew/blob/297e78734c174dd4d2585fb44049fcdb9f75357e/afew/NotmuchSettings.py#L28

I am merging this patch, thank you !
